Backpack-SwiftUI/AppSearchModal/Classes/BPKAppSearchModal.swift

```diff
@@ -0,0 +1,152 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2023 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+public struct BPKAppSearchModal: View {
+    let title: String
+    @Binding var inputText: String
+    let inputHint: String
+    let results: BPKAppSearchModalResults
+    let closeAccessibilityLabel: String
+    let onClose: () -> Void
+    
+    public init(
+        title: String,
+        inputText: Binding<String>,
+        inputHint: String,
+        results: BPKAppSearchModalResults,
+        closeAccessibilityLabel: String,
+        onClose: @escaping () -> Void
+    ) {
+        self.title = title
+        self._inputText = inputText
+        self.inputHint = inputHint
+        self.results = results
+        self.closeAccessibilityLabel = closeAccessibilityLabel
+        self.onClose = onClose
+    }
+    
+    public var body: some View {
+        VStack(spacing: .base) {
+            
+            makeNavigationBar(title: title, closeAccessibilityLabel: closeAccessibilityLabel, onClose: onClose)
+            
+            switch results {
+            case .loading(let loading):
+                BPKTextField(placeholder: inputHint, $inputText)
+                AppSearchModalLoadingView(state: loading)
+            case .content(let content):
+                BPKTextField(placeholder: inputHint, $inputText)
+                AppSearchModalContentView(state: content)
+                    .padding(.top, .md)
+            case .error(let error):
+                AppSearchModalErrorView(state: error)
+                    .padding(.horizontal, .md)
+            }
+        }
+        .padding(.horizontal, .base)
+        .padding(.top, .base)
+        .padding(.bottom, 0)
+        .background(.surfaceElevatedColor)
+    }
+    
+    func makeNavigationBar(
+        title: String,
+        closeAccessibilityLabel: String,
+        onClose: @escaping () -> Void
+    ) -> some View {
+        ZStack {
+            HStack {
+                BPKIconView(.close, size: .large)
+                    .onTapGesture(perform: onClose)
+                    .accessibilityHidden(false)
+                    .accessibilityRemoveTraits(.isImage)
+                    .accessibilityAddTraits(.isButton)
+                    .accessibilityLabel(closeAccessibilityLabel)
```

Comment:
for later in a different PR altogether (cc: @gert-janvercauteren ): I think we need to create a BPK Navigation Header or something, this is getting more and more used across screens and would be nice to have a bpk component for it!